libkbfs/block_retrieval_queue.go

```diff
@@ -276,17 +282,22 @@ func (brq *blockRetrievalQueue) Shutdown() {
 }
 
 // TogglePrefetcher allows upstream components to turn the prefetcher on or off
-func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
-	brq.prefetchMtx.Lock()
-	defer brq.prefetchMtx.Unlock()
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) (err error) {
+	var shutdownCh <-chan struct{}
+	func() {
+		brq.prefetchMtx.Lock()
+		defer brq.prefetchMtx.Unlock()
+		shutdownCh = brq.prefetcher.Shutdown()
+	}()
 	select {
-	case <-brq.prefetcher.Shutdown():
+	case <-shutdownCh:
 	case <-ctx.Done():
+		err = ctx.Err()
```

Comment:
Why isn't this `return ctx.Err()`?  Why is it the right thing to set a new prefetcher when we couldn't shut down the old one?  (If that's important, we should at least have a comment explaining why.)